wordpress-mobile/WordPress-Android
# Domain Details Screen as WebView

Fixes #19071

## Desctiption
This PR redirects to a webview when the user taps on a domain in the domain management screen. The implementation reuses the `WPWebViewActivity` with a simpified UI.

https://github.com/wordpress-mobile/WordPress-Android/assets/304044/4c5ce1c0-6614-4e96-86e2-9f2d9c80d38c

## To test:
1. Enable the Domain Management feature flag (in debug settings)
2. Open the My domains screen: Me -> Domains -> My Domains
3. Tap on a domain
4. Verify that the `domain_management_my_domains_screen_domain_tapped` event is emmited
5. Verify that the domain details open in a webview

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

6. What automated tests I added (or what prevented me from doing so)
Added testcases for the new actions in the `DomainManagementViewModelTest`

PR submission checklist:


UI Changes testing checklist:
